ui/src/app/machines/views/MachineList/MachineList.js

```diff
@@ -1,5 +1,67 @@
-import React from "react";
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
 
-const MachineList = () => <div>Machines</div>;
+import {
+  general as generalActions,
+  machine as machineActions,
+  resourcepool as resourcePoolActions,
+  scripts as scriptActions,
+  service as serviceActions,
+  tag as tagActions,
+  user as userActions,
+  zone as zoneActions
+} from "app/base/actions";
+import { machine as machineSelectors } from "app/base/selectors";
+import Col from "app/base/components/Col";
+import Loader from "app/base/components/Loader";
+import Row from "app/base/components/Row";
 
-export default MachineList;
+const DnsForm = () => {
+  const dispatch = useDispatch();
+
+  const machines = useSelector(machineSelectors.all);
+  const machinesLoaded = useSelector(machineSelectors.loaded);
+  const machinesLoading = useSelector(machineSelectors.loading);
+
+  useEffect(() => {
+    if (!machinesLoaded) {
```

Comment:
I'm not sure if this is necessary, the saga will not fire this again if the data has been fetched and this effect will only rerun if dispatch changes (which it won't in our app as we never swap out the store).

Reply:
I'll fix this up in my branch.